pkg/builtins/bitwise.go

```diff
@@ -0,0 +1,104 @@
+package builtins
+
+import (
+	"math"
+
+	"github.com/lambdaclass/cairo-vm.go/pkg/lambdaworks"
+	"github.com/lambdaclass/cairo-vm.go/pkg/vm/memory"
+	"github.com/pkg/errors"
+)
+
+const BITWISE_BUILTIN_NAME = "bitwise"
+const BITWISE_CELLS_PER_INSTANCE = 5
+const BITWISE_TOTAL_N_BITS = 251
+const BIWISE_INPUT_CELLS_PER_INSTANCE = 2
+
+type BitwiseBuiltinRunner struct {
+	base     memory.Relocatable
+	included bool
+}
+
+func BitwiseError(err error) error {
+	return errors.Wrapf(err, "Bitwise builtin error\n")
+}
+
+func ErrFeltBiggerThanPowerOfTwo(felt lambdaworks.Felt, exponent uint) error {
+	upperBound := uint64(math.Pow(2, float64(exponent)))
+	return BitwiseError(errors.Errorf("Expected felt %d to be smaller than %d", felt, upperBound))
+}
+
+func NewBitwiseBuiltinRunner(included bool) *BitwiseBuiltinRunner {
+	return &BitwiseBuiltinRunner{
+		included: included,
+	}
+}
+
+func (b *BitwiseBuiltinRunner) Base() memory.Relocatable {
+	return b.base
+}
+
+func (b *BitwiseBuiltinRunner) Name() string {
+	return BITWISE_BUILTIN_NAME
+}
+
+func (b *BitwiseBuiltinRunner) InitializeSegments(segments *memory.MemorySegmentManager) {
+	b.base = segments.AddSegment()
+}
+
+func (b *BitwiseBuiltinRunner) InitialStack() []memory.MaybeRelocatable {
+	if b.included {
+		return []memory.MaybeRelocatable{*memory.NewMaybeRelocatableRelocatable(b.base)}
+	} else {
+		return []memory.MaybeRelocatable{}
+	}
+}
+
+func (b *BitwiseBuiltinRunner) DeduceMemoryCell(address memory.Relocatable, segments *memory.Memory) (*memory.MaybeRelocatable, error) {
+	index := address.Offset % BITWISE_CELLS_PER_INSTANCE
+	if index < BIWISE_INPUT_CELLS_PER_INSTANCE {
+		return nil, nil
+	}
+
+	x_addr, err := address.SubUint(index)
+	if err != nil {
+		return nil, BitwiseError(err)
+	}
```

Comment:
Here index will never be bigger than addr.Offset so we can ignore the error (with a comment)